File: EhautX/Service/Crypto.cs
using System.Text;

namespace EhautX.Service
{
    public static class Crypto
    {
        public static string UsernameEncrypt(string username)
        {
            var sb = new StringBuilder();
            sb.Append("{SRUN3}\r\n");

            for (int i = 0; i < username.Length; ++i)
                sb.Append((char)(username[i] + 4));

            return sb.ToString();
        }

        public static string PasswordEncrypt(string password, string passwordKey)
        {
            var sb = new StringBuilder();

            for (int i = 0; i < password.Length; ++i)
            {
                char ki = (char)(password[i] ^ passwordKey[passwordKey.Length - i % passwordKey.Length - 1]);
                char _l = (char)((ki & 0x0f) + 0x36);
                char _h = (char)((ki >> 4 & 0x0f) + 0x63);
                if (i % 2 == 0)
                    sb.Append($"{_l}{_h}");
                else
                    sb.Append($"{_h}{_l}");
            }

            return sb.ToString();
        }
    }
}

File: EhautX.iOS/EhautX/Service/Crypto.swift
//
//  Crypto.swift
//  EhautTeam
//
//  Created by zh on 2019/5/26.
//  Copyright © 2019 ehaut. All rights reserved.
//

import Foundation

class Crypto {
    static func usernameEncrypt(username: String) -> String {
        var user: String = "{SRUN3}\r\n"

        for each in username {
            let e = each.asciiValue + 4
            user += e.asciiToString
        }

        return user
    }

    static func passwordEncrypt(password: String, passwordKey: String) -> String {
        var passwd: String = ""
        let passwordKeyLength = passwordKey.count
        let key = Array(passwordKey)
        var i = 0
        for each in password {
            let ki: Int = each.asciiValue ^ key[passwordKeyLength - i % passwordKeyLength - 1].asciiValue
            let _l: Int = Int((ki & 0x0F) + 0x36)
            let _h: Int = Int((ki >> 4 & 0x0F) + 0x63)
            if i % 2 == 0 {
                passwd += _l.asciiToString + _h.asciiToString
            } else {
                passwd += _h.asciiToString + _l.asciiToString
            }
            i += 1
        }

        return passwd
    }
}

extension Character {
    var asciiValue: Int {
        let s = String(self).unicodeScalars
        return Int(s[s.startIndex].value)
    }
}

extension Int {
    var asciiToString: String {
        return String(UnicodeScalar(UInt8(self)))
    }
}
